Model/AdyenInitPayments.php

```diff
@@ -200,6 +239,28 @@ protected function buildPaymentsRequest(Quote $quote, array $stateData): array
             'channel' => self::PAYMENT_CHANNEL_WEB
         ];
 
+        // Only add these parameters if the user is logged in
+        if ($isLoggedIn) {
+            $isRecurringEnabled = $this->vaultHelper->getPaymentMethodRecurringActive(
+                $paymentMethodCode,
+                $storeId
+            );
+
+            $recurringProcessingModel = $this->vaultHelper->getPaymentMethodRecurringProcessingModel(
+                $paymentMethodCode,
+                $storeId
+            );
+
+            $shopperReference = $this->requestHelper->getShopperReference($customerId, null);
+
+            $request = array_merge($request, [
+                'storePaymentMethod' => $isRecurringEnabled,
+                'shopperReference' => $shopperReference,
+                'recurringProcessingModel' => $recurringProcessingModel,
```

Comment:
If recurring is not enabled for this payment method, `recurringProcessingModel` shouldn't be sent in the request.